src/pages/home/report/ReportActionItemMessageEdit.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import {View, Pressable, Text} from 'react-native';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import ReportActionPropTypes from './ReportActionPropTypes';
+import styles from '../../../styles/styles';
+import TextInputFocusable from '../../../components/TextInputFocusable';
+import {editReportComment, saveReportActionDraft} from '../../../libs/actions/Report';
+import {scrollToIndex} from '../../../libs/ReportScrollManager';
+import toggleReportActionComposeView from '../../../libs/toggleReportActionComposeView';
+import withWindowDimensions, {windowDimensionsPropTypes} from '../../../components/withWindowDimensions';
+
+const propTypes = {
+    // All the data of the action
+    action: PropTypes.shape(ReportActionPropTypes).isRequired,
+
+    // Draft message
+    draftMessage: PropTypes.string.isRequired,
+
+    // ReportID that holds the comment we're editing
+    reportID: PropTypes.number.isRequired,
+
+    // Position index of the report action in the overall report FlatList view
+    index: PropTypes.number.isRequired,
+
+    /* Window Dimensions Props */
+    ...windowDimensionsPropTypes,
+};
+
+class ReportActionItemMessageEdit extends React.Component {
+    constructor(props) {
+        super(props);
+        this.updateDraft = this.updateDraft.bind(this);
+        this.deleteDraft = this.deleteDraft.bind(this);
+        this.debouncedSaveDraft = _.debounce(this.debouncedSaveDraft.bind(this), 1000, true);
+        this.publishDraft = this.publishDraft.bind(this);
+        this.triggerSaveOrCancel = this.triggerSaveOrCancel.bind(this);
+
+        this.state = {
+            draft: this.props.draftMessage,
+        };
+    }
+
+    /**
+     * Update the value of the draft in Onyx
+     *
+     * @param {String} newDraft
+     */
+    updateDraft(newDraft) {
+        const trimmedNewDraft = newDraft.trim();
+        this.setState({draft: trimmedNewDraft});
+        this.debouncedSaveDraft(trimmedNewDraft);
+    }
+
+    /**
+     * Delete the draft of the comment being edited. This will take the comment out of "edit mode" with the old content.
+     */
+    deleteDraft() {
+        saveReportActionDraft(this.props.reportID, this.props.action.reportActionID, '');
+        toggleReportActionComposeView(true, this.props.isSmallScreenWidth);
```

Comment:
This caused a [regression](https://github.com/Expensify/App/issues/23908) (case 2). If we delete any comment the main composer will show up, even if we are deleting a non-focused message.

What we had to do is to add a condition where we show the main composer only if we are deleting the currently focused message.